state/factory/workingset.go

```diff
@@ -195,12 +200,36 @@ func (ws *workingSet) runAction(
 			)
 		}
 		if receipt != nil {
+			isBlobTx := false // TODO: get blob type from action
+			if protocol.MustGetFeatureCtx(ctx).DisableBlobTransaction || !isBlobTx {
+				return receipt, nil
+			}
+			if err = ws.handleBlob(ctx, selp, receipt); err != nil {
+				if e := ws.Revert(si); err != nil {
+					log.L().Panic("failed to revert snapshot", zap.Error(e))
+				}
+				return nil, err
+			}
 			return receipt, nil
 		}
 	}
```

Comment:
should be added here?
```
if err = ws.handleBlob(ctx, selp, receipt); err != nil {
    return nil, err
}
```
there should not be a revert, since the blob is part of tx, if handle blob failed, the tx should be failed, and return error

Reply:
Yes, but there is a problem with this approach. If deduction of blob fee fails due to insufficient account balance, it would result in a failed block packaging, which could become a potential vulnerability.

Reply:
yeah, good thought. We can add an additional check for balance in each protocol (like transfer protocol now checks balance > amount + gas fee), or can be a common checker in `workingset`.

Reply:
added balance checking in action validation stage